src/uct/ib/base/ib_md.c

```diff
@@ -1904,4 +1929,63 @@ uct_component_t uct_ib_component = {
     .flags              = 0,
     .md_vfs_init        = uct_ib_md_vfs_init
 };
-UCT_COMPONENT_REGISTER(&uct_ib_component);
+UCT_COMPONENT_REGISTER_DEF(&uct_ib_component, ib);
+
+void UCS_F_CTOR uct_ib_init()
+{
+    uct_component_ib_ctor();
+
+#if HAVE_TL_DC
+    uct_tl_dc_mlx5_init();
+#endif
+
+#if HAVE_TL_RC
+    uct_tl_rc_verbs_init();
+#ifdef HAVE_MLX5_HW
+    uct_tl_rc_mlx5_init();
+#endif
+#endif
+
+#if HAVE_TL_UD
+    uct_tl_ud_verbs_init();
+#ifdef HAVE_MLX5_HW_UD
+    uct_tl_ud_mlx5_init();
+#endif
+#endif
+
+    uct_ib_md_ops_verbs_init();
+
+#ifdef HAVE_MLX5_DV
+    uct_ib_md_ops_dv_init();
+#if HAVE_DEVX
+    uct_ib_md_ops_devx_init();
+#endif
+#endif
+
+#if defined (HAVE_MLX5_HW) && defined (HAVE_VERBS_EXP_H)
+    uct_ib_md_ops_exp_init();
+#endif
+}
+
+void UCS_F_DTOR uct_ib_cleanup()
+{
+#if HAVE_TL_UD
+#ifdef HAVE_MLX5_HW_UD
+    uct_tl_ud_mlx5_cleanup();
+#endif
+    uct_tl_ud_verbs_cleanup();
+#endif
+
+#if HAVE_TL_RC
+#ifdef HAVE_MLX5_HW
+    uct_tl_rc_mlx5_cleanup();
+#endif
+    uct_tl_rc_verbs_cleanup();
+#endif
+
+#if HAVE_TL_DC
+    uct_tl_dc_mlx5_cleanup();
+#endif
```

Comment:
maybe define empty wrappers for these funcs in the very beginning (where you have the includes) to avoid repetition of these many #ifs?